### PR TITLE
usm: Added usm to agent status command

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Adds USM to the status agent check.

### Motivation

Getting easy understand of the state of USM (running/not running) and in case of no running, the error.

### Additional Notes

Example of a failure:
image

Example of USM is disabled:
image

Example of success:
image

### Describe how to test/QA your changes

1. Disable USM in the configuration
2. Run system probe and the agent
3. wait 15 seconds, then run `agent status` and expect to see the same output as we have under Additional Notes -> Example of USM is disabled

4. Enable USM
5. Re-run system probe
6. wait 15 seconds, then run `agent status` and expect to see the same output as we have under Additional Notes -> Example of success

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by Kaderinho — APPROVED** [review]

File: pkg/network/protocols/http/monitor.go
Comment (DanielLavie) [inline pkg/network/protocols/http/monitor.go:111]: Consider changing it to `HTTPError` or something similar, so we could easily add other protocols to the status in the near future

Comment (guyarb) [reply pkg/network/protocols/http/monitor.go:111]: the monitor should be renamed and instead of http monitor it should be USMMonitor or something similar
@p-lambert asked to wait until CO-RE support is merged before making the rename

**Review by DanielLavie — APPROVED** [review]

File: pkg/network/protocols/http/monitor.go
Comment (hmahmood) [inline pkg/network/protocols/http/monitor.go:111]: Shouldn't this just be an empty map?

File: pkg/network/tracer/tracer.go
Comment (hmahmood) [inline pkg/network/tracer/tracer.go:819]: Can't all of these errors be moved into the `http` package? You don't have to export `USMStartupError` then.

File: pkg/network/protocols/http/monitor.go
Comment (hmahmood) [inline pkg/network/protocols/http/monitor.go:115]: When is this timestamp updated? Do we intend to update it every time `GetUSMStats` is called?

File: releasenotes/notes/adding-usm-to-agent-status-check-79112d5ee853b5e5.yaml
Comment (urseberry) [inline releasenotes/notes/adding-usm-to-agent-status-check-79112d5ee853b5e5.yaml:13]:
```suggestion
    Adding Universal Service Monitoring to the Agent status check.
    Now Datadog has visibility into the status of Universal Service 
    Monitoring. Startup failures appear in the status check.
```

File: pkg/network/protocols/http/monitor.go
Comment (guyarb) [reply pkg/network/protocols/http/monitor.go:111]: why empty map? I do want to have a reason when USM is not running

File: pkg/network/tracer/tracer.go
Comment (guyarb) [reply pkg/network/tracer/tracer.go:819]: only if we move the entire function, as this is the only place that knows why USM didn't start

File: pkg/network/protocols/http/monitor.go
Comment (guyarb) [reply pkg/network/protocols/http/monitor.go:115]: every time `GetHTTPStats` is being called, it triggers `m.telemetry.log()` which updates the time

File: pkg/network/tracer/tracer.go
Comment (hmahmood) [reply pkg/network/tracer/tracer.go:819]: I meant set `USMStartupError` inside `NewMonitor`?

**Review by urseberry — APPROVED** [review]

**Review by brycekahle — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

**Pull request merged**
